modules/common/chowdsp_core/Functional/chowdsp_EndOfScopeAction.h

```diff
@@ -4,22 +4,40 @@ namespace chowdsp
 {
 /**
  * A struct that will run some code in it's destructor.
- * Inspired by: https://www.gingerbill.org/article/2015/08/19/defer-in-cpp/
+ * Inspired by: https://dev.to/pauljlucas/finally-4ke2
  */
 template <typename Func>
 struct EndOfScopeAction
 {
-    explicit EndOfScopeAction (Func&& func) : f (std::move (func)) {}
-    ~EndOfScopeAction() { f(); }
+    explicit EndOfScopeAction (Func&& func) : f (std::move (func)), invoke (true) {}
 
     EndOfScopeAction (const EndOfScopeAction&) = delete;
     EndOfScopeAction& operator= (const EndOfScopeAction&) = delete;
 
-    EndOfScopeAction (EndOfScopeAction&&) noexcept = default;
-    EndOfScopeAction& operator= (EndOfScopeAction&&) noexcept = default;
+    EndOfScopeAction& operator= (EndOfScopeAction&& other) noexcept
+    {
+        if (invoke)
+            f();
```

Comment:
Is this the correct behaviour? If I understand right, this means that if you move assign an end of scope action to an instance where `invoke` is true, then the current action will be executed before the move assignment happens. I guess this is how most "RAII" things work (e.g. `std::unique_ptr`), but I'm not sure it's the "expected behaviour" here, given that the action would executed at a time other than the "end of scope" as the user sees it?